src/components/Workspace.jsx

```diff
@@ -128,6 +131,24 @@ class Workspace extends React.Component {
     addEventListener('beforeunload', this._confirmUnload);
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
Switching to declarative (by way of a component wrapper) is way better than what I was proposing 👍 💯

Just for the sake of discussion, I'll say I don't have an issue with something like a saga being devoted to UI provided that's the *only* thing it does; my thinking is that interacting with a window actually has a lot in common with interacting with a server, and that if sagas are appropriate for the latter then they might be appropriate for the former.

**Anyway** I think the encapsulated component approach is best, *but* `react-popout` doesn't look like it's getting much in the way of support these days. [This PR](https://github.com/JakeGinnivan/react-popout/pull/32) calls out the fact that it's pinned to React 14 😞 You could roll your own?